src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1216,6 +1216,21 @@ final SQLCollation getDatabaseCollation() {
             .getLogger("com.microsoft.sqlserver.jdbc.Connection");
     private static String loggingClassName = "com.microsoft.sqlserver.jdbc.SQLServerConnection:";
 
+    /** Client process ID sent during login */
+    private static int pid = 0;
+
+    static {
+        long pidLong = 0;
+        try {
+            pidLong = ProcessHandle.current().pid();
+        } catch (NoClassDefFoundError e) { // ProcessHandle is Java 9+
+            if (loggerExternal.isLoggable(Level.FINER) && Util.isActivityTraceOn()) {
+                loggerExternal.finer(loggingClassName + " NoClassDefFoundError for ProcessHandle. ProcessId will be 0.");
+            }
+        }
+        pid = (pidLong > Integer.MAX_VALUE) ? 0 : (int)pidLong;
```

Comment:
If NoClassDefFoundError is caught, what value will pidLong be? Is it still 0 r something unexpected like Integer.MAX_VALUE or Integer.MIN_VALUE?

Reply:
should be 0? line 1223 initialized that to 0

Reply:
So in what case, pidLong would be bigger than Integer.MAX_VALUE?

Reply:
In the case where the actual process ID is simply greater than the maximum integer value. This is theoretically possible as Windows process ID's are stored as DWORDs and could go up to 2^(32 − 1).